gateway-js/src/config.ts

```diff
@@ -121,69 +126,174 @@ interface GatewayConfigBase {
   experimental_didResolveQueryPlan?: Experimental_DidResolveQueryPlanCallback;
   experimental_didFailComposition?: Experimental_DidFailCompositionCallback;
   experimental_didUpdateComposition?: Experimental_DidUpdateCompositionCallback;
+  /**
+   * @deprecated use `pollIntervalInMs` instead
+   */
   experimental_pollInterval?: number;
+  pollIntervalInMs?: number;
   experimental_approximateQueryPlanStoreMiB?: number;
   experimental_autoFragmentization?: boolean;
   fetcher?: typeof fetch;
   serviceHealthCheck?: boolean;
 }
 
-export interface RemoteGatewayConfig extends GatewayConfigBase {
+// TODO(trevor:removeServiceList)
+export interface ServiceListGatewayConfig extends GatewayConfigBase {
+  /**
+   * @deprecated: use `supergraphSdl` in its function form instead
+   */
   serviceList: ServiceEndpointDefinition[];
+  /**
+   * @deprecated: use `supergraphSdl` in its function form instead
+   */
   introspectionHeaders?:
     | HeadersInit
-    | ((service: ServiceEndpointDefinition) => Promise<HeadersInit> | HeadersInit);
+    | ((
+        service: ServiceEndpointDefinition,
+      ) => Promise<HeadersInit> | HeadersInit);
 }
 
 export interface ManagedGatewayConfig extends GatewayConfigBase {
   /**
    * This configuration option shouldn't be used unless by recommendation from
```

Comment:
Not related to this PR but — the other two options below are actually documented and real (and we don't need a comment like this on them) right?

Reply:
Correct. I think with the introduction of an additional default endpoint, we can less strongly discourage configuring this. Though we might _still_ want to discourage since the reasons for configuring are probably questionable.

Reply:
c5dcc7a3